commcare_connect/opportunity/forms.py

```diff
@@ -334,17 +335,48 @@ def __init__(self, *args, **kwargs):
             Row(Field("name")),
             Row(Field("description")),
             Row(Field("amount")),
-            Row(Field("deliver_units")),
+            Row(Field("required_deliver_units")),
+            Row(Field("optional_deliver_units")),
             Submit(name="submit", value="Submit"),
         )
 
         choices = [(deliver_unit.id, deliver_unit.name) for deliver_unit in deliver_units]
-        self.fields["deliver_units"] = forms.MultipleChoiceField(choices=choices, widget=forms.CheckboxSelectMultiple)
+        self.fields["required_deliver_units"] = forms.MultipleChoiceField(
+            choices=choices,
+            widget=forms.CheckboxSelectMultiple,
+            help_text="All of the selected Deliver Units are required for payment accrual.",
+        )
+        self.fields["optional_deliver_units"] = forms.MultipleChoiceField(
+            choices=choices,
+            widget=forms.CheckboxSelectMultiple,
+            help_text=(
+                "Any one of these Deliver Units combined with all the required "
+                "Deliver Units will accrue payment. Multiple Deliver Units can be selected."
+            ),
+            required=False,
+        )
         if PaymentUnit.objects.filter(pk=self.instance.pk).exists():
-            self.fields["deliver_units"].initial = [
-                deliver_unit.pk for deliver_unit in self.instance.deliver_units.all()
+            deliver_units = self.instance.deliver_units.all()
+            self.fields["required_deliver_units"].initial = [
+                deliver_unit.pk for deliver_unit in filter(lambda x: not x.optional, deliver_units)
+            ]
+            self.fields["optional_deliver_units"].initial = [
+                deliver_unit.pk for deliver_unit in filter(lambda x: x.optional, deliver_units)
             ]
 
+    def clean(self):
```

Comment:
Should this also validate the fact that a DeliverUnit can't be assigned to multiple PaymentUnits.

Note `payment_unit__isnull=True` in  https://github.com/dimagi/commcare-connect/pull/282/files#diff-1914cafdbf59040f7f77df353660e2634bcf3ec8e789b018d9864bc6ee7fded2R365

Reply:
This is handled in the views itself only deliver units that are not assigned to any payment units are available for the forms. So if any arbitrary deliver unit is assigned the form validation would fail as that deliver unit is not available to the form.